Sources/TuistGraph/Models/Target.swift

```diff
@@ -83,7 +83,13 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
         self.product = product
         self.destinations = destinations
         self.bundleId = bundleId
-        self.productName = productName ?? name.sanitizedModuleName
+        if let productName {
+            self.productName = productName
+        } else if product == .framework || product == .staticFramework {
+            self.productName = "$(TARGET_NAME:c99extidentifier)"
+        } else {
+            self.productName = "$(TARGET_NAME)"
+        }
```

Comment:
`productName` is used in a few different contexts within Tuist in addition to setting the underlying build setting. While Xcode can recognise variables  - sadly Tuist can't and will regress a few things within it where the `productName` is relied upon to derive various xcodeProj element / product names.

The logic needs to continue to be defined in tuist code as opposed to via variables.

Reply:
Ah makes sense. I'll revert this. It's not really related to the original issue. Anyway it's a pity that we can't use such variables. But maybe someday we will. )